internal/resolution/datasource/auth.go

```diff
@@ -0,0 +1,285 @@
+package datasource
+
+import (
+	"bytes"
+	"context"
+	"crypto/md5"
+	"crypto/rand"
+	"encoding/base64"
+	"encoding/hex"
+	"fmt"
+	"net/http"
+	"slices"
+	"strings"
+	"sync/atomic"
+)
+
+type AuthenticationMethod int
+
+const (
+	AuthBasic AuthenticationMethod = iota
+	AuthBearer
+	AuthDigest
+)
+
+type AuthenticationInfo struct {
```

Comment:
Can you add some struct level code comments to explain this struct ? It seems to be used by both npm and Maven, so it's worth explaining how this all works for multiple ecosystems. Is this just generic http auth support?

Reply:
I've added a brief top-level comment - though I'm not really sure how much detail you wanted